staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd_test.go

```diff
@@ -160,6 +163,64 @@ func TestDelete(t *testing.T) {
 	test.TestDelete(validNewCustomResource())
 }
 
+func TestGeneration(t *testing.T) {
+	// enable alpha feature CustomResourceSubresources
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CustomResourceSubresources, true)()
+
+	storage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.CustomResource.Store.DestroyFunc()
+	modifiedRno := *validNewCustomResource()
+	modifiedRno.SetGeneration(10)
+	modifiedRno.SetObservedGeneration(10)
+	ctx := genericapirequest.NewDefaultContext()
+	cr, err := createCustomResource(storage.CustomResource, modifiedRno, t)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	etcdCR, err := storage.CustomResource.Get(ctx, cr.GetName(), &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	storedCR, _ := etcdCR.(*unstructured.Unstructured)
+
+	// Generation initialization
+	if storedCR.GetGeneration() != 1 && storedCR.GetObservedGeneration() != 0 {
```

Comment:
This should be `||` instead of `&&` like below.

Reply:
Also, can you fix this for replicaset and replicationcontroller tests too?

Reply:
Done